diff --git a/lib/utils/assert.js b/lib/utils/assert.js
index 72362884f2bf87341d64663bb076705c822075bc..830fa6b66472650ea39d156218c6970ec0f4ed38 100644
--- a/lib/utils/assert.js
+++ b/lib/utils/assert.js
@@ -3,7 +3,7 @@ import { AuthError, DuplicateConditionalUI, ExperimentalFeatureNotEnabled, Inval
 let warned = false;
 function isValidHttpUrl(url, baseUrl) {
     try {
-        return /^https?:/.test(new URL(url, url.startsWith("/") ? baseUrl : undefined).protocol);
+        return /^(https?:|storyteller:)/.test(new URL(url, url.startsWith("/") ? baseUrl : undefined).protocol);
     }
     catch {
         return false;
