diff --git a/bin.cjs b/bin.cjs
index 4d22c4eb7800e3ec7c30c2c0ffcbbd490702d180..b8325fa06f3934582056e6aa597d8c86b7942700 100755
--- a/bin.cjs
+++ b/bin.cjs
@@ -58446,8 +58446,14 @@ function isMetadataYarnClassic(metadataPath) {
 }
 
 // src/services/package-manager.ts
-var PACKAGE_MANAGERS = ["npm", "bun", "pnpm", "yarn"];
-var BINARY_RUNNERS = { npm: ["npx"], bun: ["bunx", "--bun"], pnpm: ["pnpm", "dlx"], yarn: ["yarn"] };
+var PACKAGE_MANAGERS = ["npm", "bun", "pnpm", "yarn", "yarn@berry"];
+var BINARY_RUNNERS = {
+  npm: ["npx", "--yes"],
+  bun: ["bunx", "--bun", "--yes"],
+  pnpm: ["pnpm", "dlx", "--yes"],
+  yarn: ["yarn", "--yes"],
+  "yarn@berry": ["yarn", "dlx"]
+};
 var detectPackageManager = (cwd2) => Effect_exports.tryPromise({
   try: () => {
     return detect({
@@ -58462,7 +58468,7 @@ var getPackageManager = (cwd2) => Effect_exports.gen(function* () {
   if (!pm) {
     return "npm";
   }
-  const name = PACKAGE_MANAGERS.find((name2) => pm.name.startsWith(name2) || pm.agent.startsWith(name2)) ?? "npm";
+  const name = PACKAGE_MANAGERS.find((name2) => pm.agent.startsWith(name2) || pm.name.startsWith(name2)) ?? "npm";
   return name;
 });
 var getBinaryRunner = (cwd2) => Effect_exports.gen(function* () {
@@ -58514,9 +58520,8 @@ var Add = class extends Effect_exports.Service()("Add", {
         const shadcnOptions = toShadcnOptions(options3);
         const binaryRunner = yield* packageManager.getBinaryRunner(options3.cwd);
         const commandArgs = [
-          binaryRunner[1],
-          "--yes",
-          "shadcn@latest",
+          ...binaryRunner.slice(2),
+          "shadcn",
           "add",
           ...shadcnOptions,
           ...componentUrls
