diff --git a/dist/modules/src/conversion.js b/dist/modules/src/conversion.js
index cf5feb3f0da0d05798c6855373f1c7bebaad6749..ba96cb3501c53774a1a11f1b779a4b995b245a01 100644
--- a/dist/modules/src/conversion.js
+++ b/dist/modules/src/conversion.js
@@ -744,8 +744,9 @@ export class Conversion {
         let sampleRate = trackOptions.sampleRate ?? originalSampleRate;
         let needsResample = numberOfChannels !== originalNumberOfChannels
             || sampleRate !== originalSampleRate
-            || this._startTimestamp > 0
-            || firstTimestamp < 0;
+            || this._startTimestamp > 0;
+            // PATCH: see https://github.com/Vanilagy/mediabunny/issues/48
+            // || firstTimestamp < 0;
         let audioCodecs = this.output.format.getSupportedAudioCodecs();
         if (!trackOptions.forceTranscode
             && !trackOptions.bitrate
